server/punishments.js

```diff
@@ -1344,14 +1344,14 @@ Punishments.checkIp = function (user, connection) {
 			return null;
 		}
 		throw e;
-	}).then((/** @type {string} */ host) => {
+	}).then((/** @type {string|null} */ host) => {
 		user = connection.user || user;
 		if (host) user.latestHost = host;
-		Chat.hostfilter(host, user, connection);
+		Chat.hostfilter(host || "", user, connection);
```

Comment:
not a fan of this honestly, alternative is casting to String, but this is somewhat more efficient